File: src/transformers/features/protobuf/ProtobufIsEncodeTransformer.ts
import { ProtobufIsEncodeProgrammer } from "../../../programmers/protobuf/ProtobufIsEncodeProgrammer";

import { ITransformProps } from "../../ITransformProps";
import { GenericTransformer } from "../../internal/GenericTransformer";

export namespace ProtobufIsEncodeTransformer {
  export const transform = (props: ITransformProps) =>
    GenericTransformer.scalar({
      ...props,
      method: "protobuf.isEncode",
      write: ProtobufIsEncodeProgrammer.write,
    });
}

File: packages/typia/native/transform/features/protobuf/ProtobufIsEncodeTransformer.go
package protobuf

import (
	shimast "github.com/microsoft/typescript-go/shim/ast"
	nativecontext "github.com/samchon/typia/packages/typia/native/core/context"
	nativeprotobufprogrammers "github.com/samchon/typia/packages/typia/native/core/programmers/protobuf"
	nativeinternal "github.com/samchon/typia/packages/typia/native/transform/internal"
)

type protobufIsEncodeTransformerNamespace struct{}

var ProtobufIsEncodeTransformer = protobufIsEncodeTransformerNamespace{}

func (protobufIsEncodeTransformerNamespace) Transform(props nativeinternal.ITransformProps) *shimast.Node {
	return nativeinternal.GenericTransformer.Scalar(nativeinternal.GenericTransformer_IProps{
		ITransformProps: props,
		Method:          "protobuf.isEncode",
		Write: func(x nativecontext.IProgrammerProps) *shimast.Node {
			return nativeprotobufprogrammers.ProtobufIsEncodeProgrammer.Write(nativeprotobufprogrammers.ProtobufIsEncodeProgrammer_IProps(x))
		},
	})
}
